### ISSUE TITLE
Using resource in "get" doesn't populate /tag correctly

### ISSUE THREAD

**Issue by fatih** [concourse/docker-image-resource#78]
Hi,

First I have a file called `image_tag` with the following content:

```sh
123abc
```

I'm using the resource in get  and then as put in another job, a simplified example.:

```
jobs:
  - name: build
    plan:
      - task: build
        file: build.yml
      - put: my-image
        params:
          build: build
          tag: build/image_tag
  - name: deploy
    plan:
      - get: my-image
        passed: [build]
        trigger: true    
        params:
          skip-download: true
      - task: deploy
        file: deploy.yml
```

So according to the README, it says:

```
/tag: The tag of the repository that was fetched.
```

However when I fetch it I see this:

```
root@f688e206-4802-4915-4082-7ad25f7dfa2e:/tmp/build/b0d51b9f/my-image# cat tag
latest
```

But if you look at my jobs, and especially at the `put` params, you see I don't use `latest`. Instead I'm using my own tag file with the content `123abc`. 

I assume that with `get`, the content of `/tag` should also be `123abc`, but right now it's not. 

Do I miss something or is `/tag` always latest?

**Comment by ls-yann-david** [comment]
I'm currently hitting the same bug and that prevents me from finishing my CD flow 😢 

Any idea when something is going to be done for that?

**Comment by VanAxe** [comment]
After a bit of digging, this seems to be a problem _(or feature)_ with the Docker architecture. Pulling an image only pulls the used tag into the metadata available to `docker inspect`.

It seems that every tag is fundamentally associated with a _"different image"_ (notice the scare quotes). If you pull two _different tags_ that happen to be composed of all the same layers, the Docker CLI will be _nice enough_ to collate that together into the `RepoTags` metadata object, but it seems that this information is completely artificial not intrinsically available.

I've been looking the Docker API and there is no way to either get all the tags associated with a given digest, since a digest will only represent a single tag, or get all the _other_ tags also available for a given image.

So, in a nutshell, since Docker seems to only use digests internally to track collections of layers and associates a tag with a digest, and every tag and digest is unique, there is nothing available to link tags together for an identical collection of layers short of downloading all the tags and layers and checking yourself if different tags have the same layers (which I assume is what the Docker CLI does).

TL;DR, you have to track your own tag some other way. :/

This seems pretty bleak, but if someone thinks of something, I'm all ears.

**Comment by hugowetterberg** [comment]
I've solved this in my pipelines by passing the tag to a semver resource and then pass that around instead.

**Comment by chendrix** [comment]
Hi there,

Sorry this took so long to be responded to from the Concourse team but @VanAxe's https://github.com/concourse/docker-image-resource/issues/78#issuecomment-283988046 is exactly right.

Tags are not lightweight metadata attached to a digest, and the tag that's received is always the tag that's specified in your *resource definition*.

**Comment by lbathina** [comment]
@hugowetterberg: I am facing same issue. would you be able share a reference of how you solved it?

**Comment by mattdodge** [comment]
Is this still the case for this issue? I can't believe there's not a better way to push a docker image with a custom/dynamic tag and then deploy that image.

My use case is a pretty straightforward one (I think...). I want to push a docker image that has a tag of the repository's commit SHA. I would expect that if I do a get on this docker-image resource (especially immediately after the put) I can see the tag I just pushed to. But the `/tag` file always just has "latest". 

We need to rely on resources to pass information between jobs - and if the docker-image resource can't maintain information about tags, then what is the alternative? The semver option is an interesting hack but is also super inconvenient as it makes you conform your tags to some kind of major.minor.patch format. Am I missing something or doing something wrong? Or is there a better way I'm not thinking of?
